### PR TITLE
Lift out single iterators if they are leafs and tfmd is not needed.

### PR DESCRIPTION
**Pull request by baldersheim**
@havardpe PR

### CODE REVIEW

File: searchlib/src/vespa/searchlib/queryeval/weighted_set_term_blueprint.cpp [hunk 1]
```diff
@@ -96,11 +96,15 @@ SearchIterator::UP
 WeightedSetTermBlueprint::createLeafSearch(const fef::TermFieldMatchDataArray &tfmda, bool) const
 {
     assert(tfmda.size() == 1);
+    if ((_terms.size() == 1) && tfmda[0]->isNotNeeded() && _terms[0]->isLeaf()) {
```
Comment (havardpe) [inline searchlib/src/vespa/searchlib/queryeval/weighted_set_term_blueprint.cpp:99]: consider doing this as an actual optimization

File: searchlib/src/vespa/searchlib/queryeval/weighted_set_term_blueprint.cpp [hunk 2]
```diff
@@ -96,11 +96,15 @@ SearchIterator::UP
...
+        return static_cast<LeafBlueprint &>(*_terms[0]).createLeafSearch(tfmda, true);
```
Comment (havardpe) [inline searchlib/src/vespa/searchlib/queryeval/weighted_set_term_blueprint.cpp:100]: need a good comment here to explain why we cannot preserve the iterator strictness

File: searchlib/src/vespa/searchlib/queryeval/weighted_set_term_blueprint.cpp (hunk 1 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/queryeval/weighted_set_term_blueprint.cpp:99]: Turned out that was not possible due to mismatch of handle vs matchdata mismatch.

File: searchlib/src/vespa/searchlib/queryeval/weighted_set_term_blueprint.cpp (hunk 2 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/queryeval/weighted_set_term_blueprint.cpp]: Comment added. PTAL

**Review by havardpe — DISMISSED** [review]

**Pull request merged**